From: Shaoying Xu <shaoyi@amazon.com>
Date: Fri, 19 Apr 2024 18:27:55 +0000
Subject: Revert "x86/cpu: Actually turn off mitigations by default for
 SPECULATION_MITIGATIONS=n"

This reverts commit 9c09773917fbb77dff85b433e1e89123fc5fb530 merged in
v6.1.87.

Original commit turned off mitigations regardless of any architecutre if
CONFIG_SPECULATION_MITIGATIONS is disabled and this config is only defined in x86_64.
Therefore mitigations in arm64 are turned off as well. Fixes were
proposed by commit author in
https://lore.kernel.org/all/20240417001507.2264512-1-seanjc@google.com/#t
but not merged to mainline yet. AL kernel rebased to v6.1.87 will revert this
commit to unblock the release.

Signed-off-by: Shaoying Xu <shaoyi@amazon.com>
---
 kernel/cpu.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/cpu.c b/kernel/cpu.c
--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -2788,8 +2788,7 @@ enum cpu_mitigations {
 };
 
 static enum cpu_mitigations cpu_mitigations __ro_after_init =
-	IS_ENABLED(CONFIG_SPECULATION_MITIGATIONS) ? CPU_MITIGATIONS_AUTO :
-						     CPU_MITIGATIONS_OFF;
+	CPU_MITIGATIONS_AUTO;
 
 static int __init mitigations_parse_cmdline(char *arg)
 {

